lib/matplotlib/axes/_axes.py

```diff
@@ -6411,9 +6411,12 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
         """
         Plot a histogram.
 
-        Compute and draw the histogram of *x*.  The return value is a tuple
-        (*n*, *bins*, *patches*) or ([*n0*, *n1*, ...], *bins*, [*patches0*,
-        *patches1*, ...]) if the input contains multiple data.  See the
+        Computes and draws the histogram of *x*. By default, this function
+        assumes that *x* is unbinned data and bins the data as part of
+        computing the histogram. If the input is an array, then the return
```

Comment:
I'm not sure why we say "by default" here - I'm not aware that the method does anything else...

```suggestion
        Computes the histogram of the data in *x* (see `numpy.histogram`) and 
        draws the results.  If the input is an array, then the return
```

Reply:
I read the original tweet as the OP though they had to pass the histogram values.  But who knows.    

Maybe the problem here is that the first paragraph goes into the multidimensionality of *x* too soon, and that should be its own paragraph.  But the proposed change here doesn't say anything about making the bins to me (binning data is putting the data in bins, not creating the bins in the first place).

Reply:
I worry it is confusing to tell 99% of people who already know what "histogram" means that "x is assumed to be unbinned", because the knee jerk reaction is "of course it is unbinned, thats why I want to histogram it".   Perhaps better to explicitly say what the method does (computes a histogram of *x* and draws it) and then caveat if folks already have binned data ("If you have already computed a histogram, you can display using `bar` or `stairs`.  You can also pass through the drawing used by this method by passing the *bins* and *weights* as described by the *weights* parameter")